### PR TITLE
Better names in asteroid.filterbanks.transforms

### PR DESCRIPTION
**Pull request by mpariente**
As announced in #335 

I didn't add a version, but I guess it'll be the release after the next.

### CODE REVIEW

File: asteroid/filterbanks/transforms.py [hunk 1]
```diff
@@ -47,11 +47,18 @@ def mul_c(inp, other, dim: int = -2):
     return torch.cat([real1 * real2 - imag1 * imag2, real1 * imag2 + imag1 * real2], dim=dim)
 
 
-def take_reim(x, dim: int = -2):
-    return x
+def reim(x, dim: int = -2) -> Tuple[torch.Tensor]:
```
Comment (jonashaag) [inline asteroid/filterbanks/transforms.py:50]: Shouldn't this be `Tuple[torch.Tensor, torch.Tensor]`

**Comment by jonashaag** [comment]
Sorry for making this late suggestion, but since we're changing names anyway... how about moving the "Asteroid complex" functions into a submodule, like `asteroid.asteroid_complex`? If I understand correctly most/all of these functions only work on Asteroid-style complex numbers by default.

File: asteroid/filterbanks/transforms.py (hunk 1 above)
Comment (mpariente) [reply asteroid/filterbanks/transforms.py:50]: I'm not very familiar with typing yet, you're right, I'll fix it.

**Comment by mpariente** [comment]
No problem at all. 


> most/all of these functions only work on Asteroid-style complex numbers 

This is correct. Three exceptions: `compute_delta`, `concat_deltas` and `ebased_vad` 
What do you suggest exactly? What would we do with those three exceptions? 
Would `complex_nn` stay untouched?

**Comment by jonashaag** [comment]
I don't know where these would fit. Are those "transforms"? I guess yes, but many thing are "transforms", so I really don't know how to classify those, sorry!

We could move the rest to `asteroid.asteroid_complex` and add a note to the top of the module that this is only one of the three complex representations (as in `complex_nn`). And `complex_nn` stays the same.

**Comment by mpariente** [comment]
Agree that "transforms" is not informative. 
Several person already asked if we can outsource the code for asteroid filterbanks outside of asteroid and I wonder today if this wouldn't be a good opportunity. What your take on this?

**Comment by mpariente** [comment]
The only functions on which the filterbank submodule depend are `script_if_tracing` and `mixture_consistency` which are unlikely to change so I'd be ok to copy them there as well.. Not the best design but I guess it's ok. 

The advantages of this: 
- More people would use it (`torch-audiomentations` and `pyannote-audio` for example).
- More people might contribute to it? Not sure. 

And, the API is stable IMO so we don't expect to struggle to keep things compatible. 

Mabe @hbredin would be happy about that?

**Comment by hbredin** [comment]
Happy about having a standalone `asteroid.filterbanks`  package? I certainly would 💯!

**Comment by jonashaag** [comment]
Well this PR escalated quickly 😄

@mpariente I think what you suggest is reasonable (including the tiny amount of copy pasta).

**Comment by mpariente** [comment]
> Well this PR escalated quickly :smile: 

I think it's because if we'd move things to `asteroid.asteroid_complex`, making a new package out of those things would be a pain..

File: asteroid/filterbanks/transforms.py (hunk 1 above)
Comment (mpariente) [reply asteroid/filterbanks/transforms.py]: And now the linter is complaining.. 
```def chunk(self: Tensor, chunks: _int, dim: _int=0) -> Union[Tuple[Tensor, ...], List[Tensor]]: ...```

**Pull request merged**